sw/host/opentitanlib/src/otp/alert_handler.rs

```diff
@@ -0,0 +1,364 @@
+        let index = match cls {
+            AlertClass::A => 0,
+            AlertClass::B => 1,
+            AlertClass::C => 2,
+            AlertClass::D => 3,
+            AlertClass::X => bail!("Bad class"),
+        };
+
+        let mut reg = 0 as u32;
+        let reg_bits = reg.view_bits_mut::<Lsb0>();
+        match config.enabled {
+            AlertEnable::None => {}
+            AlertEnable::Enabled => {
+                reg_bits.set(ALERT_HANDLER_CLASSA_CTRL_SHADOWED_LOCK_BIT as usize, true);
+                reg_bits.set(ALERT_HANDLER_CLASSA_CTRL_SHADOWED_EN_BIT as usize, true);
+            }
+            AlertEnable::Locked => {
+                reg_bits.set(ALERT_HANDLER_CLASSA_CTRL_SHADOWED_EN_BIT as usize, true)
+            }
+        }
+
+        match config.escalate {
+            AlertEscalate::Phase0 => {
+                reg_bits.set(ALERT_HANDLER_CLASSA_CTRL_SHADOWED_EN_E0_BIT as usize, true)
+            }
+            AlertEscalate::Phase1 => {
+                reg_bits.set(ALERT_HANDLER_CLASSA_CTRL_SHADOWED_EN_E0_BIT as usize, true);
+                reg_bits.set(ALERT_HANDLER_CLASSA_CTRL_SHADOWED_EN_E1_BIT as usize, true);
+            }
+            AlertEscalate::Phase2 => {
+                reg_bits.set(ALERT_HANDLER_CLASSA_CTRL_SHADOWED_EN_E0_BIT as usize, true);
+                reg_bits.set(ALERT_HANDLER_CLASSA_CTRL_SHADOWED_EN_E1_BIT as usize, true);
+                reg_bits.set(ALERT_HANDLER_CLASSA_CTRL_SHADOWED_EN_E2_BIT as usize, true);
+            }
+            AlertEscalate::Phase3 => {
+                reg_bits.set(ALERT_HANDLER_CLASSA_CTRL_SHADOWED_EN_E0_BIT as usize, true);
+                reg_bits.set(ALERT_HANDLER_CLASSA_CTRL_SHADOWED_EN_E1_BIT as usize, true);
+                reg_bits.set(ALERT_HANDLER_CLASSA_CTRL_SHADOWED_EN_E2_BIT as usize, true);
+                reg_bits.set(ALERT_HANDLER_CLASSA_CTRL_SHADOWED_EN_E3_BIT as usize, true);
+            }
+            AlertEscalate::None => {}
+        }
+
+        self.class[index] = reg;
+        self.class_regs[index].accum_thresh = config.accum_thresh;
+        self.class_regs[index].timeout_cyc = config.timeout_cyc;
+        self.class_regs[index].phase_cycs = config.phase_cycs;
+
+        Ok(())
+    }
+}
+
+trait CRC32Add {
+    fn crc32_add(self, diegst: &mut Digest<u32>);
+}
+
+impl CRC32Add for u32 {
+    fn crc32_add(self, digest: &mut Digest<u32>) {
+        digest.update(self.to_le_bytes().as_slice())
+    }
+}
+
+impl<T: CRC32Add, const N: usize> CRC32Add for [T; N] {
+    fn crc32_add(self, digest: &mut Digest<u32>) {
+        self.map(|v| v.crc32_add(digest));
+    }
+}
+
+impl CRC32Add for AlertClassRegs {
+    fn crc32_add(self, digest: &mut Digest<u32>) {
+        self.regwen.crc32_add(digest);
+        self.ctrl.crc32_add(digest);
+        self.accum_thresh.crc32_add(digest);
+        self.timeout_cyc.crc32_add(digest);
+        self.phase_cycs.crc32_add(digest);
+    }
+}
+
+impl CRC32Add for AlertRegs {
```

Comment:
Did I miss it or the integration with the opentitantool will be covered in a separate PR? If so, please keep in mind that:
```
alert_config_crc32_for_lc_state = crc32 ^ lc_state ^ kErrorOk;
```